### PR TITLE
Test optimized Parquet writer in Hive product tests

### PR DESCRIPTION
**Pull request by Justin-Kwan**
**Description**

Issue: https://github.com/trinodb/trino/issues/6738

- Test parquet optimized reader (re-using existing parquet reader tests)
- Test failures are fixed by skipping Timestamp datatypes for the optimizer reader, pending: https://github.com/trinodb/trino/issues/5357

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java [hunk 1]
```diff
@@ -240,6 +241,7 @@
         return Stream.of(storageFormats())
                 // everything but Avro supports nanoseconds
                 .filter(format -> !"AVRO".equals(format.getName()))
+                .filter(format -> !format.getSessionProperties().containsKey("hive.experimental_parquet_optimized_writer_enabled"))
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:244]: - The comment above says that all but Avro support nanoseconds, but new parquet writer clearly does not. Update the comment.

- The check here should also check that a) name = "PARQUET" and b) session property value is "true".

- Please file an issue for this, and add TODO comment here, with a link to the issue.

Comment (Justin-Kwan) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:244]: @findepi Would it make sense to have the issue: https://github.com/trinodb/trino/issues/5357 also responsible for updating this test case (skipping nanoseconds for optimized reader) to verify correctness? Either way, I will create tag it in the code.

Comment (Justin-Kwan) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:244]: https://github.com/trinodb/trino/issues/8715

Comment (Justin-Kwan) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:244]: Also, which extra session property are you referring to?

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@Justin-Kwan i added `tests:hive` label to give this test a run against all tested Hive versions. Please check the build results when ready, and let me know.

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**